lib/manageiq/schema/schema_statements.rb

```diff
@@ -42,6 +46,11 @@ def drop_miq_metric_table_inheritance(table, inherit_from)
         execute("ALTER TABLE #{quoted_table} NO INHERIT #{quoted_inherit}", 'Drop table inheritance')
       end
 
+      def change_miq_metric_sequence(table, inherit_from)
+        drop_sequence(table)
+        change_column_default(table, :id, -> { "nextval('#{inherit_from}_id_seq')" })
```

Comment:
Have you tested this change with `rake db:schema:load`?  This was part of what I was making sure didn't cause an error with this commit from my [original PR](https://github.com/ManageIQ/manageiq-schema/pull/504):

- https://github.com/ManageIQ/manageiq-schema/commit/ca2a5dbac4d1f33461a81bc97bf94f2f6096c24a

If this does work, does that mean we can remove `column_spec_for_primary_key`, or is this part to augment it?

Reply:
I did not...I only tried `rake db:reset`...I'll give that a shot.

Reply:
Just tried with `db:schema:load` and it works :+1:

---

I tried removing/adjusting column_spec_for_primary_key, but there are multiple problems.  If I straight remove it, the problem is that Rails tries to create the id column with

```sql
"id" bigserial DEFAULT nextval('metric_rollups_base_id_seq'::regclass) NOT NULL PRIMARY KEY
```

The problem there is that there is a sequence declared, but `bigserial` _also_ implies adding a default sequence, so there are 2 sequences, this yielding this error: `ERROR:  multiple default values specified for column "id" of table "metric_rollups_01"`.  The fix is to instead use `bigint` when you declare a sequence, and I hacked that in, but then I get a different error: `ERROR:  relation "metric_rollups_base_id_seq" does not exist`, which is because the tables are in alphabetical order and not in topological order.

I'll investigate changing the ordering, but I still am not sure how to change the bigserial to bigint without a manual hack.

Reply:
Yeah, the bigger problem I'm having is forcing rails to use `bigint` over `bigserial`.  Under the covers, the Rails type is specified, and `:bigint` auto-maps to `bigserial`.

I think we have to leave the changes as is.  That is, keep `column_spec_for_primary_key` to remove the default, and let it create the sequence via `bigserial`, then we run this method which removes the sequence and puts the base sequence in.

Reply:
Okay, cool.  I just was a bit concerned with that because it looked very similar to what was in the commit message I linked, so if we are good here with `db:schema:load`, no worries then.

Kinda sucks it requires such a round about way of handling it, but what we are dong is a bit different from a standard Rails deployment.

Anyway, I am good with this change, just wanted to make sure.  Thanks!